src/networkrequest.cpp

```diff
@@ -777,6 +777,34 @@ void NetworkRequest::replyFinished() {
     return;
   }
 
+  if (m_reply->error() == QNetworkReply::HostNotFoundError && isRedirect()) {
+    QUrl brokenUrl = m_reply->url();
+
+    if (brokenUrl.host().isEmpty() && !m_redirectedUrl.isEmpty()) {
+      QUrl url = m_redirectedUrl.resolved(brokenUrl);
+
+#ifdef MVPN_DEBUG
+      logger.debug()
+          << "QT6 redirect bug! The current URL is broken because it's not "
+             "resolved using the latest HTTP redirection as base-URL";
+      logger.debug() << "Broken URL:" << brokenUrl.toString();
+      logger.debug() << "Latest redirected URL:" << m_redirectedUrl.toString();
+      logger.debug() << "Final URL:" << url.toString();
+#endif
+
+      m_request = QNetworkRequest(url);
+      m_request.setHeader(QNetworkRequest::ContentTypeHeader,
+                          "application/json");
+      m_request.setAttribute(QNetworkRequest::RedirectPolicyAttribute,
+                             QNetworkRequest::NoLessSafeRedirectPolicy);
+
+      m_reply = nullptr;
```

Comment:
I'm concerned that this may cause a memory leak of the failed `QNetworkReply` now that `NetworkRequest::maybeDeleteLater()` will do nothing if `m_reply` is null.

Reply:
We do not leak it because at line 868 we call `m_reply->setParent(this);`